jerry-core/parser/js/js-parser-expr.c

```diff
@@ -888,6 +888,11 @@ parser_reparse_as_common_identifier (parser_context_t *context_p, /**< context *
                                      parser_line_counter_t start_line, /**< start line */
                                      parser_line_counter_t start_column) /**< start column */
 {
+  if (context_p->token.lit_location.type != LEXER_IDENT_LITERAL)
+  {
+    parser_raise_error (context_p, PARSER_ERR_IDENTIFIER_EXPECTED);
+  }
```

Comment:
I would add an explanation that `context_p->token.lit_location.char_p` is showing the character after the string start, so it is not suitable for reparsing.

There is an `if` below:
```
if (context_p->token.type != LEXER_LITERAL
      || context_p->token.lit_location.type != LEXER_IDENT_LITERAL)
```
I think we don't need the `context_p->token.lit_location.type != LEXER_IDENT_LITERAL` after this change.

Reply:
Good idea!